slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -52,6 +52,40 @@ ::grpc::Status SLAMServiceImpl::GetPosition(ServerContext *context,
     return grpc::Status::OK;
 }
 
+::grpc::Status SLAMServiceImpl::GetPositionNew(
+    ServerContext *context, const GetPositionNewRequest *request,
+    GetPositionNewResponse *response) {
+    cartographer::transform::Rigid3d global_pose;
+    {
+        std::lock_guard<std::mutex> lk(viam_response_mutex);
+        global_pose = latest_global_pose;
+    }
+
+    // Set pose for our response
+    Pose *myPose = response->mutable_pose();
+    myPose->set_x(global_pose.translation().x());
+    myPose->set_y(global_pose.translation().y());
+    myPose->set_z(global_pose.translation().z());
+
+    // Set extra for our response (currently stores quaternion)
+    google::protobuf::Struct *q;
+    google::protobuf::Struct *extra = response->mutable_extra();
+    q = extra->mutable_fields()->operator[]("quat").mutable_struct_value();
+    q->mutable_fields()->operator[]("real").set_number_value(
+        global_pose.rotation().w());
+    q->mutable_fields()->operator[]("imag").set_number_value(
+        global_pose.rotation().x());
+    q->mutable_fields()->operator[]("jmag").set_number_value(
+        global_pose.rotation().y());
+    q->mutable_fields()->operator[]("kmag").set_number_value(
+        global_pose.rotation().z());
+
+    // Set component_reference for our response
+    response->set_component_reference(camera_name);
```

Comment:
Where is `set_component_reference` defined?

Reply:
`set_component_reference` is defined in our [slam proto](https://github.com/viamrobotics/api/blob/main/proto/viam/service/slam/v1/slam.proto#L104) and takes the `camera_name` from the [sensors](https://github.com/viamrobotics/slam/blob/main/slam-libraries/viam-cartographer/src/slam_service/config.cc#L25) input. Also def worth noting that if sensors are not included b/c of leaving it out of the config in offline mode, then an empty string will be set for the cartographer case

Reply:
It is a grpc definition from proto